src/swap/FeeInfoBottomSheet.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import BottomSheet, { BottomSheetRefType } from 'src/components/BottomSheet'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import TokenDisplay from 'src/components/TokenDisplay'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { AppFeeAmount, SwapFeeAmount } from 'src/swap/types'
+
+interface Props {
+  forwardedRef: React.RefObject<BottomSheetRefType>
+  appFee?: AppFeeAmount
+  crossChainFee?: SwapFeeAmount
+  networkFee?: SwapFeeAmount
+  fetchingSwapQuote: boolean
+}
+
+function FeeAmount({
+  fee,
+  showMaxAmount,
+  isLoading,
+}: {
+  fee: SwapFeeAmount | undefined
+  showMaxAmount?: boolean
+  isLoading: boolean
+}) {
+  const { t } = useTranslation()
+
+  const displayAmount = showMaxAmount ? fee?.maxAmount : fee?.amount
+  const isFeeZero = displayAmount && displayAmount.isZero()
+  const loadingText = t('loading')
+  const fallbackText = t('unknown')
+  const zeroFeeText = t('swapScreen.transactionDetails.swapFeeWaived')
+
+  return (
+    <Text style={styles.feeAmountText}>
+      {isLoading ? (
+        loadingText
```

Comment:
i spent >30 minutes on this but couldn't get the styles working for some reason....i think it's a good idea but i also think that the chances of users seeing this state is small (probably only likely if they are waiting a long time for a quote, but even so it may not be likely that they want info about the fees during the wait). so for now i'm giving up on it but maybe i'll return to it at some point 😅